downstreamadapter/eventcollector/event_collector.go

```diff
@@ -649,23 +665,43 @@ func (d *DispatcherStat) unregisterDispatcher(eventCollector *EventCollector) {
 	}
 }
 
-// FIXME: Implement this method.
-func (d *DispatcherStat) resetDispatcher(eventCollector *EventCollector) {
+func (d *dispatcherStat) pauseChangefeed(eventCollector *EventCollector) {
 	d.eventServiceInfo.RLock()
 	defer d.eventServiceInfo.RUnlock()
 
 	if d.eventServiceInfo.serverID == "" || !d.eventServiceInfo.readyEventReceived {
-		log.Panic("should not happen: reset dispatcher before receiving ready signal")
+		// Just ignore the request if the dispatcher is not ready.
+		return
 	}
```

Comment:
It looks this can be split to a method

Reply:
Good advice, I will do it in a separate PR.